docs/api-docs/ModelConfigPM.md
---
title: ModelConfigPM
---

# ModelConfigPM

::: src.simple_model.ModelConfigPM


docs/getting-started/configuration.md
---
title: Configuration
---

# ⚙️ Configuration

[**`templates/configs/config.yml`**](https://github.com/bybatkhuu/model.python-template/blob/main/templates/configs/config.yml):

```yaml
simple_model:                                       # Just an example to group the configs (Not necessary)
  models_dir: "models"                              # Directory where the models are saved
  model_name: "linear_regression.v0.0.1-250101"     # Name of the model as sub-directory
  threshold: 0.5                                    # Threshold for similarity check
```

## 🌎 Environment Variables

[**`.env.example`**](https://github.com/bybatkhuu/model.python-template/blob/main/.env.example):

```sh
# ENV=LOCAL
# DEBUG=false
# TZ=UTC
# PYTHONDONTWRITEBYTECODE=0
```


CHANGELOG.md
# Changelog

## v3.0.11 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.11 -->



**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.10...v3.0.11

## v3.0.10 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.10 -->



**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.9...v3.0.10

## v3.0.9 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.9 -->

## What's Changed
### 👷 CI/CD
* Dev by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/27


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.8...v3.0.9

## v3.0.7 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.7 -->

## What's Changed
### 🐛 Fixes
* :broom: Remove version check step from build workflow by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/25


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.6...v3.0.7

## v3.0.6 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.6 -->

## What's Changed
### 💬 Other
* :green_heart: Add version check step to build workflow by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/24


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.5...v3.0.6

## v3.0.4 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.4 -->



**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.3...v3.0.4

## v3.0.4 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.4 -->



**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.3...v3.0.4

## v3.0.3 (2025-08-23)

<!-- Release notes generated using configuration in .github/release.yml at v3.0.3 -->



**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v3.0.2...v3.0.3

## v2.0.0 (2025-06-18)

<!-- Release notes generated using configuration in .github/release.yml at v2.0.0 -->

## What's Changed
### 💥 Breaking Changes
* :boom: update environment configuration and dependencies by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/20


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v1.0.0...v2.0.0

## v1.0.0 (2024-12-26)

<!-- Release notes generated using configuration in .github/release.yml at v1.0.0 -->

## What's Changed
### 📝 Documentation
* Refactor README and update logging configuration by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/19
### 💬 Other
* Update pre-commit configuration and dependencies by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/18


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v0.0.4...v1.0.0

## v0.0.4 (2024-12-20)

<!-- Release notes generated using configuration in .github/release.yml at v0.0.4 -->

## What's Changed
### 🐛 Fixes
* Refactor model directory handling and update documentation by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/17
### 📝 Documentation
* Enhance documentation headers with emoji for improved readability by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/13
* Update README and installation documentation for improved cloning instructions by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/15
* Update README for clarity and usage instructions by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/16
### 🔨 Scripts
* Update VSCode settings and clarify Makefile target names by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/14


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v0.0.3...v0.0.4

## v0.0.3 (2024-11-12)

<!-- Release notes generated using configuration in .github/release.yml at v0.0.3 -->

## What's Changed
### 📝 Documentation
* Update documentation and mkdocs configuration by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/10
* Improve error messages and add diagram generation script by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/11
* Update documentation and add diagram generation scripts by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/12


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v0.0.2...v0.0.3

## v0.0.2 (2024-11-08)

<!-- Release notes generated using configuration in .github/release.yml at v0.0.2 -->

## What's Changed
### 📝 Documentation
* Update documentation, improve scripts, and modify project structure by @bybatkhuu in https://github.com/bybatkhuu/model.python-template/pull/9


**Full Changelog**: https://github.com/bybatkhuu/model.python-template/compare/v0.0.1...v0.0.2


src/simple_model/config.py
from pydantic import BaseModel, constr, Field, ConfigDict


class ExtraBaseModel(BaseModel):
    model_config = ConfigDict(extra="allow")


class ModelConfigPM(ExtraBaseModel):
    """Pydantic model for SimpleModel config.

    Attributes:
        models_dir (str  ): Directory where models are stored. Defaults to "models".
        modelName  (str  ): Name of the model which is also the sub-directory name.
                                Defaults to "linear_regression.v0.0.1-250101".
        threshold  (float): Threshold value for the similarity score. Defaults to None.
    """

    models_dir: constr(strip_whitespace=True) = Field(  # type: ignore
        default="./models", min_length=2, max_length=1023
    )
    modelName: constr(strip_whitespace=True) = Field(  # type: ignore
        default="linear_regression.v0.0.1-250101",
        min_length=2,
        max_length=128,
        alias="model_name",
    )
    threshold: float = Field(default=0.5, ge=0.0, le=1.0)


__all__ = ["ModelConfigPM"]
